src/types/profile.js

```diff
@@ -626,6 +628,11 @@ export type Thread = {|
   resourceTable: ResourceTable,
   nativeSymbols: NativeSymbolTable,
   jsTracer?: JsTracerTable,
+  // If present and true, this thread was launched for a private browsing
+  // session only.
+  isPrivateBrowsing?: boolean,
+  // If present, the number represents the container this thread was loaded in.
+  userContextId?: number,
```

Comment:
Actually in the thread structure, it is also absent when the threads had no special extra attribute. For example:
* for a normal tab in a normal session, the 2 properties will be missing.
* for a tab loaded in a container but otherwise non-private, we'd have both values
* for a tab loaded in a private window but not in a container, we'd have both values

I've expanded the comment to explain this better.

Reply:
Actually... I'll make them always present in the processed format :-)